src/Backend/Core/Engine/Model.php

```diff
@@ -643,7 +643,7 @@ public static function getURLForBlock($module, $action = null, $language = null)
 
         // still no page id?
         if ($pageIdForURL === null) {
-            return self::getURL(404);
+            return self::getURL(404, $languag);
```

Comment:
Since language is a parameter of the method, passing it on is logical. (it will use the current language by default)